Cannot find module '/node_modules/react/jsx-runtime' imported from '/node_modules/next-mdx-remote/dist/index.js'

I am using the NextJS latest version `12.1.1` and facing this issue. 

Inside the `/node_modules/next-mdx-remote/dist/index.js` it import `react/jsx-runtime`

```js
import * as runtime from 'react/jsx-runtime.js';
```

Which create the issue but if I changed it with actual path it works.

```js
import * as runtime from 'react/cjs/react-jsx-runtime.production.min.js';
``` 

But how to solve it in deployment stage? Like vercel also show the same issue when  deploying. 

```shell
> Build error occurred
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/vercel/path0/node_modules/react/jsx-runtime' imported from /vercel/path0/node_modules/next-mdx-remote/dist/index.js
```

Isn't `next-mdx-remote` compatible with `nextJS@12.1.1`?

--- a/__tests__/integration.test.tsx
+++ b/__tests__/integration.test.tsx
@@ -21,13 +21,15 @@ describe('hydration', () => {
     const result = readOutputFile('basic', 'index')
     const $ = cheerio.load(result)
 
+    const htmlOutput = $('#__next').html()
+
     // server renders correctly
-    expect($('#__next').html()).toContain(`<h1>foo</h1><h1>Headline</h1>
-<p>hello <!-- -->jeff</p><button>Count: <!-- -->0</button>
-<p class=\"context\">Context value: \"<!-- -->foo<!-- -->\"</p>
-<p>Some <strong class=\"custom-strong\">markdown</strong> content</p>
-<div class=\"alert alert-warning g-type-body\"><p>Alert</p></div>
-<div>I am a dynamic component.</div>`)
+    expect(htmlOutput).toContain(`<h1>foo</h1><h1>Headline</h1>
+<!-- --><p>hello <!-- -->jeff<!-- --></p><button>Count: <!-- -->0<!-- --></button>
+<!-- --><p class=\"context\">Context value: \"<!-- -->foo<!-- -->\"<!-- --></p>
+<!-- --><p>Some <!-- --><strong class=\"custom-strong\">markdown</strong> content<!-- --></p>
+<!-- --><div class=\"alert alert-warning g-type-body\"><p>Alert</p></div>
+<!-- --><div>I am a dynamic component.</div>`)
   })
 
   test('rehydrates correctly in browser', async () => {

--- a/package.json
+++ b/package.json
@@ -36,12 +36,12 @@
     "esbuild-jest": "^0.5.0",
     "husky": "^4.3.6",
     "jest": "^27.3.1",
-    "next": "^12.0.10",
+    "next": "^12.1.5",
     "prettier": "^2.4.1",
     "pretty-quick": "^3.1.1",
     "puppeteer": "^10.4.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
     "release": "^6.3.0",
     "rmfr": "^2.0.0",
     "rollup": "^2.59.0",
@@ -91,8 +91,8 @@
   ],
   "license": "MPL-2.0",
   "peerDependencies": {
-    "react": ">=16.x <=17.x",
-    "react-dom": ">=16.x <=17.x"
+    "react": ">=16.x <=18.x",
+    "react-dom": ">=16.x <=18.x"
   },
   "engines": {
     "node": ">=14",
@@ -104,6 +104,7 @@
   },
   "scripts": {
     "build": "rm -rf ./dist && rollup -c rollup.config.js",
+    "postbuild": "cp ./src/jsx-runtime.cjs dist",
     "lint": "prettier --write \"./**/*.{js,jsx,json}\"",
     "release:major": "release major && npm publish",
     "release:minor": "release minor && npm publish",

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -12,7 +12,12 @@ export default [
       dir: './dist',
       format: 'es',
     },
-    external: ['react', '@mdx-js/react', 'react/jsx-runtime.js'],
+    external: [
+      'react',
+      '@mdx-js/react',
+      'react/jsx-runtime',
+      './jsx-runtime.cjs',
+    ],
     plugins: [
       ts({
         tsconfig: './tsconfig.json',

--- a/src/index.tsx
+++ b/src/index.tsx
@@ -1,6 +1,6 @@
 import './idle-callback-polyfill'
 import React, { useEffect, useState, useMemo } from 'react'
-import * as runtime from 'react/jsx-runtime.js'
+import { jsxRuntime } from './jsx-runtime.cjs'
 import * as mdx from '@mdx-js/react'
 import { MDXRemoteSerializeResult } from './types'
 
@@ -71,7 +71,7 @@ export function MDXRemote({
     // first we set up the scope which has to include the mdx custom
     // create element function as well as any components we're using
     const fullScope = Object.assign(
-      { opts: { ...mdx, ...runtime } },
+      { opts: { ...mdx, ...jsxRuntime } },
       { frontmatter },
       scope
     )

--- a/src/jsx-runtime.cjs
+++ b/src/jsx-runtime.cjs
@@ -1,0 +1,8 @@
+/**
+ * Allow jsx-runtime to be successfully imported from either React 17 or React 18.
+ *
+ * Inspired by the approach here: https://github.com/contentlayerdev/contentlayer/blob/main/packages/next-contentlayer/src/hooks/jsx-runtime.cjs
+ */
+const jsxRuntime = require('react/jsx-runtime')
+
+module.exports.jsxRuntime = jsxRuntime

--- a/src/jsx-runtime.cjs.d.ts
+++ b/src/jsx-runtime.cjs.d.ts
@@ -1,0 +1,3 @@
+import * as jsxRuntime from 'react/jsx-runtime'
+
+export { jsxRuntime }

